service/history/execution/mutable_state_util.go

```diff
@@ -381,8 +383,11 @@ func CopyWorkflowExecutionInfo(sourceInfo *persistence.WorkflowExecutionInfo) *p
 
 // CopyActivityInfo copies ActivityInfo
 func CopyActivityInfo(sourceInfo *persistence.ActivityInfo) *persistence.ActivityInfo {
-	details := make([]byte, len(sourceInfo.Details))
-	copy(details, sourceInfo.Details)
+	var details []byte
+	if sourceInfo.Details != nil {
+		details = make([]byte, len(sourceInfo.Details))
+		copy(details, sourceInfo.Details)
+	}
```

Comment:
nit: could be a one liner
```suggestion
details := append(make([]int, 0, len(sourceInfo.Details)), sourceInfo.Details...)
```

Reply:
tbh we should probably just lean on x/exp/slices whenever possible: https://pkg.go.dev/golang.org/x/exp/slices#Clone